idpi/src/operators/flexpart.py

```diff
@@ -0,0 +1,118 @@
+# Third-party
+import cfgrib
+import numpy as np
+import xarray as xr
+import yaml
+from operators.omega_slope import omega_slope
+from operators.time_operators import time_rate
+
+
+class ifs_data_loader:
+    """Class for loading data from ifs and convert conventions to COSMO."""
+
+    def __init__(self, field_mapping_file: str):
+        with open(field_mapping_file) as f:
+            self._field_map = yaml.safe_load(f)
+
+    def open_ifs_to_cosmo(self, datafile: str, fields: list[str]):
+        ds = {}
+
+        read_keys = ["pv", "NV"]
+        ifs_multi_ds = cfgrib.open_datasets(
+            datafile,
+            backend_kwargs={"read_keys": read_keys},
+            encode_cf=("time", "geography", "vertical"),
+        )
+
+        for f in fields:
+            ds[f] = self._get_da(self._field_map[f]["ifs"]["name"], ifs_multi_ds)
+            if "cosmo" in self._field_map[f]:
+                ufact = self._field_map[f]["cosmo"].get("unit_factor")
+
+                if ufact:
+                    ds[f] *= ufact
+
+        return ds
+
+    def _get_da(self, field, dss):
+        for ds in dss:
+            if field in ds:
+                return ds[field]
+
+
+def load_flexpart_data(fields, loader, datafile):
+    ds = loader.open_ifs_to_cosmo(datafile, fields)
+
+    ds["U"] = ds["U"].sel(hybrid=slice(40, 60))
+    ds["V"] = ds["V"].sel(hybrid=slice(40, 60))
+    ds["ETADOT"] = ds["ETADOT"].sel(hybrid=slice(1, 60))
+    ds["T"] = ds["T"].sel(hybrid=slice(40, 60))
+    ds["QV"] = ds["QV"].sel(hybrid=slice(40, 60))
+
+    return ds
+
+
+def append_pv(ds):
+    """Compute ak, bk (weights that define the vertical coordinate) from pv."""
```

Comment:
@petrabaumann how is fieldextra extracting ak, bk ? 
here we are assuming that ak, bk is the same for all fields, so it just takes it from U. 
It is not a good implementation since it does not even check whether U is present. But I wanted to know first how fieldextra does it before fixing it